src/mjolnir/pbfgraphparser.cc

```diff
@@ -2081,6 +2087,10 @@ void PBFGraphParser::ParseNodes(const boost::property_tree::ptree& pt,
   // Create OSM data. Set the member pointer so that the parsing callback methods can use it.
   graph_callback callback(pt, osmdata, intersections_file, shapes_file, true);
 
+  // Read the OSMData to files if the tile_dir is set.
+  if (!tile_dir.empty())
```

Comment:
this seems very imprecise to me. is there a way to tell it to read the temp files if they havent already been read instead of checking that the directory string isnt empty? basically have osmdata_ lazy load itself

Reply:
specifically i think it would be better if we did something like:

```c++
if(!osmdata_.is_initialized())
  osmdata_.read_from_temp_files(pt.get<string>("mjolnir.tile_dir"));
```